src/itensor.jl

```diff
@@ -1277,28 +1266,26 @@ end
 
 -(A::ITensor) = itensor(-tensor(A))
 
-function (A::ITensor{N} + B::ITensor{N}) where {N}
+function (A::ITensor + B::ITensor)
+  if ndims(A) == 0 && ndims(B) > 0
+    return copy(B)
+  end
+  if ndims(B) == 0 && ndims(A) > 0
+    return copy(A)
+  end
```

Comment:
Is the above logic to replace the emptyITensor behavior? This may be a good way to go but needs some thought.

Reply:
Yes.

Reply:
It looks like it is to replace the `ITensor{Any}` behavior as I discussed in this comment: https://github.com/ITensor/ITensors.jl/pull/591/files#r594604222. Agreed that we need to think about this. For one, this seems like it would be incorrect if you did `ITensor(2) + ITensor(i, j)`, for example. For now we may only want this behavior if `ndims(A) == 0 && isempty(A)` (where by `isempty(A)` I mean that `store(A) isa Empty`, I'm not sure if that is the current definition of `isempty(::ITensor)`).